drivers/flash/soc_flash_mcux.c

```diff
@@ -204,36 +207,54 @@ static int flash_mcux_read(const struct device *dev, off_t offset,
 	 */
 	addr = offset + priv->pflash_block_base;
 
-#ifdef CONFIG_CHECK_BEFORE_READING
-  #ifdef CONFIG_SOC_LPC55S36
-	/* Validates the given address range is loaded in the flash hiding region. */
-	rc = FLASH_IsFlashAreaReadable(&priv->config, addr, len);
-	if (rc != kStatus_FLASH_Success) {
-		rc = -EIO;
-	} else {
-		/* Check whether the flash is erased ("len" and "addr" must be word-aligned). */
-		rc = FLASH_VerifyErase(&priv->config, ((addr + 0x3) & ~0x3),  ((len + 0x3) & ~0x3));
-		if (rc == kStatus_FLASH_Success) {
-			rc = -ENODATA;
+	if (IS_ENABLED(CONFIG_CHECK_BEFORE_READING)) {
+		/*
+		 * Ensure the area is readable, since a direct access may cause faults on erased
+		 * or otherwise unreadable pages. Emulate erased pages, return other errors.
+		 */
+		if (IS_ENABLED(CONFIG_SOC_LPC55S36)) {
```

Comment:
@pillo79 it looks like we may need more "#ifdefery", because some older NXP platforms don't define the `FLASH_Read` API at all. I think we probably need to use `#ifdef` here. Sorry for the poor guidance